fuel-relayer/src/relayer/syncing.rs

```diff
@@ -0,0 +1,88 @@
+use ethers_core::types::U256;
+
+use super::*;
+
+#[cfg(test)]
+mod tests;
+
+struct Status {
+    starting_block: U256,
+    current_block: U256,
+    highest_block: U256,
+}
+
+pub async fn wait_if_eth_syncing<P>(
+    eth_node: &P,
+    sync_call_freq: Duration,
+    sync_log_freq: Duration,
+) -> anyhow::Result<()>
+where
+    P: Middleware<Error = ProviderError> + 'static,
+{
+    tracing::info!("Waiting for the Ethereum endpoint to finish syncing");
+    let mut start = tokio::time::Instant::now();
+    let mut loop_time = tokio::time::Instant::now();
+    while let SyncingStatus::IsSyncing {
+        starting_block,
+        current_block,
+        highest_block,
+    } = get_status(eth_node).await?
+    {
+        if start.elapsed() > sync_log_freq {
+            let status = Status {
+                starting_block,
+                current_block,
+                highest_block,
+            };
+            start = tokio::time::Instant::now();
+            tracing::info!(
+                "Waiting for the Ethereum endpoint to finish syncing. \nStatus {}",
+                status
+            );
+        }
+        tokio::time::sleep(sync_call_freq.saturating_sub(loop_time.elapsed())).await;
+        loop_time = tokio::time::Instant::now();
+    }
+    Ok(())
+}
+
+async fn get_status<P>(eth_node: &P) -> anyhow::Result<SyncingStatus>
+where
+    P: Middleware<Error = ProviderError> + 'static,
+{
+    match eth_node.syncing().await {
+        Ok(s) => Ok(s),
```

Comment:
[nit] this match could be replaced by using anyhow context instead: https://docs.rs/anyhow/latest/anyhow/trait.Context.html